Bounteous-Inc/acs-aem-commons
# Site-wide component property support

This is an "alpha" state feature (not yet vetted for production use).

Need to test/verify:
- Ensure that properties can be sent out for translation
- Ensure that Reference component can work with global configs
- Ensure that path fields within global configs are updated if referenced paths are updated
- Ensure that merging of dialogsitewide.xml from resourceSuperType is working

Future Features and Enhancements:
- Only display the sitewide properties button for components that define a _cq_dialogsitewide node
  — Alternatively show a "no sitewide configs" message for components that do not have them
- Move the sitewide properties button on the authoring toolbar next to the wrench icon for the standard properties dialog
- Ensure authoring page refreshes when sitewide props are updated, since multiple component instances may be affected
- Flush dispatcher for all pages within the site using a component when site-wide properties are updated.
- Add support for site-wide properties that are not specific to a component type

This feature is enabled by adding a cq:infoProvider to your page component, referencing com.adobe.acs.commons.wcm.impl.SitewideComponentPropsPageInfoProvider.  See documentation at https://docs.adobe.com/docs/en/aem/6-1/develop/components/pageinfo.html for adding nodes to cq:infoProvider.
